### ISSUE TITLE
Drastically different behavior between TF1 and TF2

### ISSUE THREAD

**Issue by danaugrs** [tensorflow/tensorflow#29099]
I've noticed drastically different behavior of the following code between _2.0.0-alpha0_ and _1.13.1_.
```python
import numpy as np
from tensorflow.keras.datasets import mnist
from tensorflow.keras.utils import to_categorical
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.optimizers import Adam

(x_train_raw, y_train_raw), (x_test_raw, y_test_raw) = mnist.load_data()
x_train = x_train_raw.reshape(60000, 784)
x_test = x_test_raw.reshape(10000, 784)
y_train = to_categorical(y_train_raw)
y_test = to_categorical(y_test_raw)

basic_net = Sequential([
    Dense(36, activation='relu', input_shape=(784,)),
    Dense(10, activation='softmax')
])

W_bio = np.load('W_bio.npy')

basic_net.layers[0].set_weights([W_bio.transpose(), basic_net.layers[0].get_weights()[1]])
basic_net.layers[0].trainable = False
basic_net.compile(optimizer=Adam(lr=0.0001), metrics=['accuracy'], loss='categorical_crossentropy')
basic_net.fit(x=x_train, y=y_train, epochs=15)
```

In _2.0.0-alpha0_ the accuracy of the network consistently reaches >80% in the first few epochs. In _1.13.1_ the accuracy consistently reaches only <20% after all 15 epochs. What is going on?

Here is the file with weights: [W_bio.npy](https://drive.google.com/file/d/12O0XE98jl_yYmiBbZMWyLMaeZiHH-FXH/view?usp=sharing)

**Comment by danaugrs** [comment]
The difference in behavior has something to do with using ReLU. If instead the activation function used for the hidden layer is sigmoid then the behavior becomes the same in TF1 and TF2 (<20% accuracy).

**Comment by muddham** [comment]
@danaugrs I hope you are using ReLU for the hidden layers now. Can you please let us know if we have to keep this issue open

**Comment by danaugrs** [comment]
I think you misunderstood my last comment. I was just helping whoever is going to debug this difference in behavior between _2.0.0-alpha0_ and _1.13.1_ by pointing out that it _probably_ lies in the ReLU implementation.

**Comment by robieta** [comment]
The accuracy difference doesn't seem to be as severe now, but it is definitely still present (~65% vs. ~85%) However switching to distribution strategy (which changes the execution path) restores the accuracy in 2.0. We're standardizing on the latter path, so this should be fixed before the 2.0 release. In the mean time something like:

```
with tf.distribute.OneDeviceStrategy("GPU:0").scope():
  basic_net = Sequential([
    Dense(36, activation='relu', input_shape=(784,)),
    Dense(10, activation='softmax')
  ])
  
  basic_net.compile(optimizer=Adam(lr=0.0001), metrics=['accuracy'], loss="categorical_crossentropy")
  basic_net.fit(x=x_train, y=y_train, epochs=15)
```
will do the right thing. Thanks for the report, and for the fantastic repro.

cc @tomerk @qlzh727 @omalleyt12 @karmel

**Comment by jvishnuvardhan** [comment]
@danaugrs I think the issue was resolved in `tf-nightly` and `tf-nightly-2.0-preview`. I ran your code in both the versions and the accuracy after 15 Epochs is `78.57%` and `78.29` with `TF2.0` and `tf-nightly (TF.1.15)`. 

Here is [gist](https://colab.sandbox.google.com/gist/jvishnuvardhan/d57e90aed84f74b7c422bccb23abf48b/tfnightly20preview_29099_keras_perfomance_relu.ipynb) with `TF2.0` and [here is the gist](https://colab.sandbox.google.com/gist/jvishnuvardhan/415672186e6fcadd6093198dc33be64e/tfnightly_29099_keras_perfomance_relu.ipynb) with `tf-nightly`.

I am closing the issue as it was resolved recent nightly. Please feel free to open it if the issue persists again. Thanks!
